src/nemos/glm.py

```diff
@@ -1158,6 +1159,112 @@ def __sklearn_clone__(self) -> GLM:
+        Parameters
+        ----------
+        filename :
+            The name of the file where the model parameters will be saved. The file will be saved in `.npz` format.
+
+        Examples
+        --------
+        >>> import nemos as nmo
+
+        >>> # Create a GLM model with specified parameters
+        >>> solver_args = {"stepsize": 0.1, "maxiter": 1000, "tol": 1e-6}
+        >>> model = nmo.glm.GLM(
+        ...     regularizer="Ridge",
+        ...     regularizer_strength=0.1,
+        ...     observation_model="Gamma",
+        ...     solver_name="BFGS",
+        ...     solver_kwargs=solver_args,
+        ... )
+
+        >>> # Print the model parameters
+        >>> print(model.get_params())
+
+        >>> # Save the model parameters to a file
+        >>> model.save_params("model_params.npz")
+
+        >>> # Load the model from the saved file
+        >>> model = nmo.load_model("model_params.npz")
+
+        >>> # Print the parameters of the loaded model
+        >>> # This should match the original model parameters
+        >>> print(model.get_params())
+        """
+
+        # initialize saving dictionary
+        fit_attrs = {}
+
+        fit_attrs["coef_"] = self.coef_
+        fit_attrs["intercept_"] = self.intercept_
+
+        string_attrs = ["inverse_link_function"]
+
+        super().save_params(filename, fit_attrs, string_attrs)
+
+    @classmethod
+    def _load_from_dict(
+        cls, saved_attrs: dict[str, Any], mapping_dict: Optional[dict] = None
+    ) -> GLM:
+        """
+        Load a GLM model from a dictionary of saved attributes.
+
+        This private method is designed to be called internally by the function load_model.
+
+        Parameters
+        ----------
+        saved_attrs :
+            Dictionary containing the saved attributes of the model.
+
+        mapping_dict :
+            Optional dictionary to map custom attribute names to their actual objects.
+
+        Returns
+        -------
+        model :
+            A GLM instance with the loaded parameters.
+        """
+
+        # Get the parameter names for the class
+        model_params_names = cls._get_param_names()
+
+        saved_attrs["inverse_link_function"] = saved_attrs["observation_model"][
+            "params"
+        ]["inverse_link_function"]
+        saved_attrs["observation_model"] = saved_attrs["observation_model"]["class"]
+
+        # Filter only valid parameters for the model constructor
+        parameter_config = {
+            k: v for k, v in saved_attrs.items() if k in model_params_names
+        }
+
+        # Create the model instance
+        model = cls(**parameter_config)
+
+        # Set inverse_link_function if available
+        try:
+            if hasattr(model.observation_model, "inverse_link_function"):
+                model.observation_model.inverse_link_function = str(
+                    saved_attrs["inverse_link_function"]
+                )
+        except KeyError:
+            pass
```

Comment:
this section is confusing to me -- let's talk through this more, I feel like it can be simplified.